src/main/java/Parser.java

```diff
@@ -0,0 +1,101 @@
+import java.io.FileNotFoundException;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.Scanner;
+
+/**
+ * Deals with user commands.
+ */
+public class Parser {
+    protected static Scanner sc = new Scanner(System.in);
+
+    /**
+     * Checks user input and catches exceptions.
+     *
+     * @param storage Storage of data.
+     * @param tasks TaskList of tasks.
+     */
+    public static void run(Storage storage, TaskList tasks) {
+        while (true) {
+            try {
+                String userInput = sc.nextLine();
+
+                // Save data when user says bye
+                if (userInput.equals("bye")) {
+                    storage.saveData(tasks);
+                    Ui.sayBye();
+                    break;
+                }
+                Parser.startResponse(userInput, tasks);
+            } catch (IllegalArgumentException e) {
+                System.out.println("\nOh no Flamingo! I need more information to do that.\n");
+            } catch (IndexOutOfBoundsException e) {
+                System.out.println("\nOh no Flamingo! The task does not exist.\n");
+            } catch (NullPointerException e) {
+                System.out.println("\nOh no Flamingo! There is nothing in the list.\n");
+            } catch (FileNotFoundException e) {
+                System.out.println("\nOh no Flamingo! I cannot load the save data!\n");
+            } catch (Exception e) {
+                System.out.println("\nOh no Flamingo! I didn't understand that.\n");
+            }
+        }
+    }
+
+    /**
+     * Makes sense of the user input.
+     *
+     * @param userInput User input.
+     * @param tasks TaskList of tasks.
+     * @throws Exception If user input is unknown.
+     */
+    public static void startResponse(String userInput, TaskList tasks) throws Exception {
```

Comment:
Maybe you wanna change the if -else statement to switch statement to look neater?